File: BubbleSort.java
import java.util.Scanner;

public class BubbleSort {
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);
        int[] a = new int[5];
        System.out.println("Enter the elements of the array: ");
        for (int i = 0; i<a.length; i++){
            a[i] = scan.nextInt();
        }
        System.out.println("Array before sorting: ");
        print(a);
        long start = System.nanoTime();
        bubble(a);
        long end = System.nanoTime();
        long execution = end - start;
        System.out.println("Array after sorting: ");
        print(a);
        System.out.println("Execution time: " + execution + " nanoseconds");
    }
    static void print(int[] a){
        for (int i : a) {
            System.out.print(i + " ");
        }
        System.out.println();
    }
    public static void bubble(int[] a){
        int x = a.length;
        for(int i = 0; i<x-1; i++){
            int flag = 0;
            for (int j = 0; j<x-1; j++){
                if (a[j]> a[j+1]){
                    int temp = a[j+1];
                    a[j+1] = a[j];
                    a[j] = temp;
                    flag++;
                }
            }
            if (flag == 0){
                break;
            }
        }
    }
}

File: bubbleSort.c
#include <stdio.h>
void bubblesort(int arr[], int n)
{

    for (int i = 0; i < n - 1; i++)
    {
        int isSwaped = 0;
        for (int j = 0; j < n - i - 1; j++)
        {
            if (arr[j] > arr[j + 1])
            {
                int temp = arr[j];
                arr[j] = arr[j + 1];
                arr[j + 1] = temp;
                isSwaped = 1;
            }
        }
        if (isSwaped = 0)
        {
            break;
        }
    }
}

int main()
{
    int arr[] = {90,60,50,20,30};
    int n = sizeof(arr) / sizeof(arr[0]);
    bubblesort(arr, n);
    for (int i = 0; i < 5; i++)
    {
        printf("%d ", arr[i]);
    }

    return 0;
}